zerver/tests/test_signup.py

```diff
@@ -3257,6 +3257,26 @@ def _assert_redirected_to(self, result: HttpResponse, url: str) -> None:
         self.assertEqual(result.status_code, 302)
         self.assertEqual(result["LOCATION"], url)
 
+    def signup(self, *,
+        email = "newguy@zulip.com": str,
+        password = "newpassword": str,
+        realm=get_realm("zulip"): Realm,
```

Comment:
Evaluating a function definition in kwargs in Python doesn't work how one would expect; you'll want to do this as `if realm is not None`.  (It calls it just once at import time, not once per time you call the function)

Reply:
OK